Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -1068,6 +1068,11 @@ void RenderElement::styleDidChange(StyleDifference diff, const RenderStyle* oldS
     } else if (diff == StyleDifference::LayoutPositionedMovementOnly)
         setNeedsPositionedMovementLayout(oldStyle);
 
+    if (isOutOfFlowPositioned() && oldStyle && oldStyle->isOriginalDisplayBlockType() != style().isOriginalDisplayBlockType()) {
+        if (CheckedPtr ancestor = container())
```

Comment:
I think we want containing block here.

Reply:
That's a good point.

Is `RenderObject::markContainingBlocksForLayout` incorrect? It seems like it walks up the container chain, and would mark more intermediate ancestors than is necessary (especially for position:fixed).

Reply:
what we need is the in-flow containing block, which is usually the parent, except when the parent is an inline box e.g. `<div><span><div out-of-flow>`